cmake/modules/SwiftSupport.cmake

```diff
@@ -0,0 +1,94 @@
+
+# Returns the current architecture name in a variable
+#
+# Usage:
+#   get_swift_host_arch(result_var_name)
+#
+# If the current architecture is supported by Swift, sets ${result_var_name}
+# with the sanitized host architecture name derived from CMAKE_SYSTEM_PROCESSOR.
+function(get_swift_host_arch result_var_name)
```

Comment:
Rather than hard-code each architecture, we can ask the compiler what the module triple is based on either the default target triple for the toolchain or if someone is trying to cross-compile, based on `CMAKE_Swift_COMPILER_TARGET`:

```cmake
if(NOT SwiftFoundation_MODULE_TRIPLE)
  # TODO: This logic should migrate to CMake once CMake supports installing swiftmodules
  set(module_triple_command "${CMAKE_Swift_COMPILER}" -print-target-info)
  if(CMAKE_Swift_COMPILER_TARGET)
    list(APPEND module_triple_command -target ${CMAKE_Swift_COMPILER_TARGET})
  endif()
  execute_process(COMMAND ${module_triple_command} OUTPUT_VARIABLE target_info_json)
  string(JSON module_triple GET "${target_info_json}" "target" "moduleTriple")
  set(SwiftFoundation_MODULE_TRIPLE "${module_triple}" CACHE STRING "swift module triple used for installed swiftmodule and swiftinterface files")
  mark_as_advanced(SwiftFoundation_MODULE_TRIPLE)
endif()
```

https://github.com/etcwilde/swift/blob/e10563e336f2fdd7bde5b4b5f1c2a181aa2fe024/stdlib/cmake/modules/PlatformInfo.cmake#L16-L26